src/core/thread/neighbor_table.cpp

```diff
@@ -304,8 +307,11 @@ void NeighborTable::Signal(Event aEvent, const Neighbor &aNeighbor)
     case kChildRemoved:
         Get<Notifier>().Signal(kEventThreadChildRemoved);
 #if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
-        Get<DuaManager>().UpdateChildDomainUnicastAddress(static_cast<const Child &>(aNeighbor),
-                                                          Mle::ChildDuaState::kRemoved);
+        if (Get<ChildTable>().Contains(aNeighbor))
```

Comment:
Is it valid to have a neighbor not in child table? Should we assert?

Reply:
Same here. We can assert

Reply:
Did not add assert here. Since we do assert (under same `kChildRemove`) in same method above.